py/server/deephaven/experimental/iceberg.py

```diff
@@ -135,7 +207,7 @@ def snapshots(self, table_identifier: str) -> Table:
 
         return self.j_object.listSnapshotsAsTable(table_identifier)
 
-    def read_table(self, table_identifier: str, instructions: Optional[IcebergInstructions] = None, snapshot_id: Optional[int] = None) -> Table:
+    def read_table(self, table_identifier: str, instructions: Optional[IcebergInstructions] = None, snapshot_id: Optional[int] = None) -> IcebergTable:
         """
```

Comment:
As discussed, we might want to document the table definition of  what `snapshots()` returns, and link it to the question of  where `snapshot_id` comes from.